src/distributions/slice.rs

```diff
@@ -9,9 +9,39 @@
 use core::num::NonZeroUsize;
 
 use crate::distributions::{Distribution, Uniform};
+use crate::Rng;
 #[cfg(feature = "alloc")]
 use alloc::string::String;
 
+#[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
+const _: () = assert!(false, "unsupported pointer width");
```

Comment:
Why not use `compile_error!` here? Also, why can't we use `UniformSize::U32` on 16-bit targets?

Reply:
1. I forgot about it — I'll switch to `compile_error!`
2. Dependencies don't compile and it seems like no-one cares much about 16-bit (#1468), so there's no point wasting time here.